compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUOps.cpp

```diff
@@ -237,15 +237,34 @@ LogicalResult CoalescedGatherDMAOp::verify() {
     return emitOpError("input type must either be a tensor or a memref");
   }
 
-  if (hasTensor) {
-    if (!isa<RankedTensorType>(getIndices().getType()) ||
-        !isa<RankedTensorType>(getSource().getType())) {
-      return emitOpError("all operands must be tensors when init is a tensor");
+  // Check indices if present
+  if (Value indices = getIndices()) {
```

Comment:
See https://llvm.org/docs/CodingStandards.html#commenting
```suggestion
  // Check indices if present.
  if (Value indices = getIndices()) {
```